CMakeLists.txt

```diff
@@ -844,6 +844,29 @@ configure_file ( fluidsynth.pc.in
 install ( FILES ${CMAKE_BINARY_DIR}/fluidsynth.pc
 	DESTINATION ${LIB_INSTALL_DIR}/pkgconfig )
 
+# Exported targets for cmake: find_package(FluidSynth)
+# when installed, use CMAKE_PREFIX_PATH=fluidsynth-prefix;...
+# to use the build directory directly set the FluidSynth_DIR variable instead.
+
+# targets in the build directory
+export(EXPORT FluidSynthTargets
+   FILE "${CMAKE_CURRENT_BINARY_DIR}/FluidSynthTargets.cmake"
+   NAMESPACE FluidSynth::
+)
+
+include(CMakePackageConfigHelpers)
+write_basic_package_version_file(
+    FluidSynthConfigVersion.cmake
+    VERSION ${LIB_VERSION_INFO}
+    COMPATIBILITY SameMajorVersion
+)
+
+configure_file(FluidSynthConfig.cmake.in FluidSynthConfig.cmake @ONLY)
```

Comment:
```suggestion
configure_package_config_file(
    "${CMAKE_CURRENT_SOURCE_DIR}/FluidSynthConfig.cmake.in"
    "${CMAKE_CURRENT_BINARY_DIR}/FluidSynthConfig.cmake"
    INSTALL_DESTINATION ${LIB_INSTALL_DIR}/cmake/fluidsynth
)
```

Reply:
This is the **orthodox** way to generate the *config.cmake files, but it has the problem that only supports the installed package use case. I've changed the template, only removing the `include(FluidSynthConfigVersion.cmake)` because it is not necessary.

My idea is to generate a single file `FluidSynthConfig.cmake` in the build directory that works for both the installed and for using the build directory directly.  The included file `FluidSynthTargets.cmake` is generated in the build directory by 
~~~
export(EXPORT FluidSynthTargets ...)
~~~

And the file with the same name installed is generated by
~~~
install(EXPORT FluidSynthTargets ...)
~~~

See also: https://cliutils.gitlab.io/modern-cmake/chapters/install/installing.html

Reply:
Oh, ok. Then perhaps add a short comment, which points out that `configure_package_config_file` would only support the installed package use case.

Reply:
done